### ISSUE TITLE
subsetting problem with DimensionalData

### ISSUE THREAD

**Issue by b-r-hamilton** [ggebbie/AlgebraicArrays.jl#10]
If I try to subset using indices that include `end`, I get some weird behavior, e.g. 
```
using AlgebraicArrays, DimensionalData
v = zeros((X(1:5), Y(1:3)), :VectorArray) 
v[:, 1:end-1]
``` 
returns the following 
```
julia> v[:, 1:end-1]
╭────────────────────────────────────────────────────────────────────────────╮
│ 5×0 DimArray{Quantity{Float64, 𝐓, Unitful.FreeUnits{(yr,), 𝐓, nothing}},2} │
├────────────────────────────────────────────────────────────────────────────┴ dims ┐
  ↓ X Sampled{Int64} 1:5 ForwardOrdered Regular Points,
  → Y Sampled{Int64} 1:0 ForwardOrdered Regular Points
└──────────────────────────────────────────────────────────────────────────────┘
 ↓ →
 1
 2
 3
 4
 5
============================
*operating algebraically as*
Quantity{Float64, 𝐓, Unitful.FreeUnits{(yr,), 𝐓, nothing}}[]
```
Weirdly, this is okay for the first dimension... 
```
julia> v[1:end-1, :]
╭─────────────────────────╮
│ 4×3 DimArray{Float64,2} │
├─────────────────────────┴────────────────────────────────────────────── dims ┐
  ↓ X Sampled{Int64} 1:4 ForwardOrdered Regular Points,
  → Y Sampled{Int64} 1:3 ForwardOrdered Regular Points
└──────────────────────────────────────────────────────────────────────────────┘
 ↓ →  1    2    3
 1    0.0  0.0  0.0
 2    0.0  0.0  0.0
 3    0.0  0.0  0.0
 4    0.0  0.0  0.0
============================
*operating algebraically as*
12-element Vector{Float64}:
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0
 0.0

```

Problem also occurs for something like `v[:, 2:end]`

**Comment by b-r-hamilton** [comment]
Temporary fix is just to replace `end` with `size(v)[2]` but that feels like a cop-out!

**Comment by ggebbie** [comment]
Thanks a lot for posting the working example and workaround!  

Indeed it seems like a reasonable goal to get `begin` and `end` to work.

There are other indexing issues that I worked on but sometimes was not able to solve. I documented some of them with the `fails` notation like here: https://github.com/ggebbie/AlgebraicArrays.jl/blob/5ab25ac7ed6498fed2cceb3b2e065a63a8aeefd9/test/test_DimensionalData.jl#L64
